### PR TITLE
Feature.terminal truecolor

### PR DESCRIPTION
**Pull request by jneen**
Add support for truecolor terminals!

### CODE REVIEW

**Comment by jneen** [comment]
Try it out with `./bin/rougify -f terminal-truecolor some-file`

**Comment by jneen** [comment]
To compare:

image

vs

image

File: lib/rouge/cli.rb [hunk 1]
```diff
@@ -200,9 +200,14 @@ def self.doc
         yield %[                            delimiters. implies --escape]
       end
 
+      # TODO: find a better way to do this?
+      def self.supports_truecolor?
+        ENV['TERM'] == 'xterm' || %w(24bit truecolor).include?(ENV['COLORTERM'])
```
Comment (pyrmont) [inline lib/rouge/cli.rb]: It doesn't look like there's an especially good way to detect this in Ruby. The best I could find was [this test](https://github.com/janlelis/paint/blob/879c0a40d05563e57e7c091d2d5a96c3f5cc82a4/lib/paint.rb#L172-L193) in the Paint gem. What do you think?

File: lib/rouge/cli.rb [hunk 2]
```diff
@@ -200,9 +200,17 @@ def self.doc
...
 
-      # TODO: find a better way to do this?
+      # There is no consistent way to do this, but this is used elsewhere,
+      # and we provide explicit opt-in and opt-out with $COLORTERM
       def self.supports_truecolor?
-        ENV['TERM'] == 'xterm' || %w(24bit truecolor).include?(ENV['COLORTERM'])
+        return true if %w(24bit truecolor).include?(ENV['COLORTERM'])
+        return false if ENV['COLORTERM'] && ENV['COLORTERM'] =~ /256/
+
+        if RbConfig::CONFIG['host_os'] =~ /mswin|mingw/
```
Comment (pyrmont) [inline lib/rouge/cli.rb]: Sorry, I didn't pick up on this in my earlier comment but although part of Ruby, this needs `require 'rbconfig'` to be pulled in properly (see [this write up](https://idiosyncratic-ruby.com/42-ruby-config.html)).

**Pull request merged**